src/main/java/org/jenkinsci/plugins/gitlab/GitlabMergeRequestWrapper.java

```diff
@@ -349,6 +350,22 @@ public GitlabNote createNote(String message, boolean shouldClose, boolean should
 
     }
 
+    public GitlabCommitStatus createCommitStatus(String status, String targetUrl) {
+        try {
+            GitlabAPI api = builder.getGitlab().get();
+            GitlabMergeRequest mergeRequest = api.getMergeRequest(project, id);
+            GitlabCommit latestCommit = getLatestCommit(mergeRequest, api);
+
+            if (latestCommit != null) {
+                return api.createCommitStatus(project, latestCommit.getId(), status, mergeRequest.getSourceBranch(), "Jenkins", targetUrl, null);
```

Comment:
I'm new to the code for this plugin but I'm wondering if there is a race condition here. If a Merge Request is created and there is a subsequent commit, wouldn't this call incorrectly set the commit status on the latest commit? getLatestCommit returns the latest commit of the merge request, not the commit that is being built.

Pretty cool feature though. Looking forward to it.